### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by demix007**
## In this milestone, we implemented the following:

- [ ] Added unit tests for all the methods in all the classes, the methods include:

    💠 Book
    💠Person
    💠Student
    💠Teacher
    💠Classroom
    💠Rental

### CODE REVIEW

**Review by TadesseDev — COMMENTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑🛑
yPkfDFf

 - [ ] I see you have an empty readme file, This will affect your professionality as a developer. We need to use a professional readme file in all our projects and for that [Microverse provides a Readme template](https://github.com/microverseinc/readme-template) we can customize and use it. I suggest you do the same before the next review request.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  before :each do
+    @person1 = Person.new(12, name: 'Little Smith', parent_permission: false)
+    @person2 = Person.new(500, name: 'Elrond HalfElf')
+    @book1 = Book.new('Game of Thrones', 'George Martin')
+    @book2 = Book.new('The Songs of Achilles', 'Madelaine Miller')
+    @book3 = Book.new('The Lord of the Rings', 'Tolkien')
+    @rental = Rental.new(Time.now, @person1, @book1)
+  end
+  context 'When creating a new Person' do
+    it 'should have the following attributes (id, age, name, and parent_permission, rentals)' do
```
Comment (rustam-tolipov) [inline spec/person_spec.rb:15]: **You did a great job! I like all the implementations!!!** 

- [ ] However, we have the last test that needs to be written. You need to write a test for the name if we do not provide a name it should return us the default value "Unknown". Kindly add a test for that. Thank you!

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @demix007, and @RashidMenhas,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

You are free to remove this line from your readme:

> NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish.
You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @demix007 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer. 💯 

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by demix007** [comment]
## Hello @MahmoudBakr23 👋🏼 ,

Thank you for taking out time to review and approve this PR. You are amazing 💯

**Pull request merged**